recipes-bsp/u-boot/u-boot-iot2050.inc

```diff
@@ -48,16 +48,10 @@ dpkg_runbuild_prepend() {
     export TEE=/usr/lib/optee-os/iot2050/tee-pager_v2.bin
 }
 
-dpkg_build_sstate_prepare_append() {
-    ln -f ${S}/${SPI_FLASH_IMG} -t ${DPKG_SSTATE}
-}
-
-dpkg_build_sstate_finalize_append() {
-    ln -f ${DPKG_SSTATE}/${SPI_FLASH_IMG} -t ${S}
-}
-
 do_deploy() {
-    cp -f "${S}/${SPI_FLASH_IMG}" "${DEPLOY_DIR_IMAGE}/${SPI_FLASH_DEPLOY_IMG}"
+    dpkg --fsys-tarfile ${WORKDIR}/${PN}_${PV}_${PACKAGE_ARCH}.deb | \
+        tar xOf - ./usr/lib/u-boot/${MACHINE}/${SPI_FLASH_IMG} \
+        > "${DEPLOY_DIR_IMAGE}/${SPI_FLASH_DEPLOY_IMG}"
```

Comment:
Looks like this was not tested against master. Please do so.

Reply:
```
ERROR: u-boot-iot2050-pg2-2021.04-r0 do_deploy: Execution of '/build/tmp/work/iot2050-debian-arm64/u-boot-iot2050-pg2/2021.04-r0/temp/run.do_deploy.30011' failed with exit code 2:
dpkg-deb: error: failed to read archive '/build/tmp/work/iot2050-debian-arm64/u-boot-iot2050-pg2/2021.04-r0/u-boot-iot2050-pg2_2021.04_arm64.deb': No such file or directory
tar: This does not look like a tar archive
tar: ./usr/lib/u-boot/iot2050/flash.bin: Not found in archive
tar: Exiting with failure status due to previous errors
WARNING: exit code 2 from a shell command.
```

Reply:
Looking at it, this may actually be a bug in the existing recipe, just exposed now. Still, patches should be tested!

Reply:
Nope, debian package name is `u-boot-${MACHINE}` and stays like this, independent of the recipe name (which controls the variant). A bit special in this case. So, please adjust.

Reply:
I seem to miss this error due to the strange thing: running `./kas-container build kas-iot2050-example.yml` completed without errors and I didn't find logs for 'deploy' - it simply was not executed.

When I've run the shell and try to build u-boot in it - yes, I caught the same problem. Already fixed and will be sent a bit later.

I'm trying to understand why skipping deploy task could happen. Probably, removing dpkg_build_state_* stuff from the recipe could somehow make an impact.

Reply:
Investigation through `bitbage -g` shows that `u-boot-iot2050-pg2.do_deploy` is not in task tree when building any image. So, flash.bin is never deployed when building any of `iot2050-image-*`.

Reply:
That's true for disk image building which only happens to build u-boot for the purpose of u-boot-iot2050-config. The case above was triggered by building one of the firmware images.